pingcap/tidb
# add windows build support

### What problem does this PR solve? 
When building tidb on windows, go build retrun with some linux signal errors:
`# command-line-arguments
src\github.com\pingcap\tidb\tidb-server\main.go:441:34: undefined: syscall.SIGUSR1
src\github.com\pingcap\tidb\tidb-server\main.go:446:14: undefined: syscall.SIGUSR1 
` 

### What is changed and how it works?
add windows build support!
Change SIGUSR1 from go system control to user control by define custom signal:
`sigUsr1  := syscall.Signal(0x1e)`

### Check List 

Tests 


 - Manual test (add detailed scripts or steps below)
1. Running tidb-server with MockStore
2. kill -30 $pid
3. Custom signal still can handle dump goroutine stack request.

Code changes

Side effects


Related changes
 - Need to be included in the release note
Support windows build, but build on windows should only be use in developer environment.
